src/language-js/printer-estree.js

```diff
@@ -5285,22 +5285,79 @@ function isObjectType(n) {
 // eg; `describe("some string", (done) => {})`
 function isTestCall(n) {
   const unitTestRe = /^(skip|(f|x)?(it|describe|test))$/;
+
+  /*
+  * The below check makes sure that it('very long', async(()=>{//code}))
+  * async(()) doesn't break
+  */
+
+  if (n.arguments.length === 1) {
+    if (n.callee.type === "Identifier" && n.callee.name === "async") {
```

Comment:
We should also check its parent to be a test call if this is specific to tests.

Reply:
@ikatyang do you mean adding a check below at https://github.com/ad1992/prettier/blob/7bd1e8004627544b401a6188b3edb6ec79af8b0a/src/language-js/printer-estree.js#L967
```js
const isCalleeAsync = n.type === "Identifier" && n.callee.name === "async";
  if ((isCalleeAsync && isTestCall(parent)) || !isCalleeAsync) {
    return concat([
      isNew ? "new " : "",
      path.call(print, "callee"),
      optional,
      path.call(print, "typeParameters"),
      concat(["(", join(", ", path.map(print, "arguments")), ")"])
    ]);
  }
```

Reply:
No, I meant something like this in `isTestCall`:

```diff
-if (n.callee.type === "Identifier" && n.callee.name === "async")
+if (n.callee.type === "Identifier" && n.callee.name === "async" && isTestCall(parent))
```